## Practice yourself with `babynames`



Transfer the code from the previous section to a new dataset.

This practice file (without answers) is available in Colab as:
tinyurl.com/babyNamesColab. The answers are in
tinyurl.com/babyNamesColabAnswers.

1.  Download this file of baby names to your PC as `babynames.csv`:
    [tinyurl.com/babyNamesPractice](https://tinyurl.com/babyNamesPractice)

2.  Import the file from your `Downloads` directory to a data frame named
    `baby`.



In [1]:
import pandas as pd
import matplotlib.pyplot as plt
baby = pd.read_csv("../data/babynames.csv")

1.  Print the data frame to make sure the import worked - how many
    names does this data set contain?



In [1]:
<<baby>>
print(baby)

#+begin_example
           year      name   percent   sex
   0       1880      John  0.081541   boy
   1       1880   William  0.080511   boy
   2       1880     James  0.050057   boy
   3       1880   Charles  0.045167   boy
   4       1880    George  0.043292   boy
   ...      ...       ...       ...   ...
   257995  2008  Carleigh  0.000128  girl
   257996  2008     Iyana  0.000128  girl
   257997  2008    Kenley  0.000127  girl
   257998  2008    Sloane  0.000127  girl
   257999  2008   Elianna  0.000127  girl

   [258000 rows x 4 columns]
   #+end_example

1.  There is only one numeric column, the `percent` (of records that



In [1]:
<<baby>>
print(baby.iloc[:,2].describe())

: count    258000.000000
   : mean          0.000894
   : std           0.003016
   : min           0.000026
   : 25%           0.000081
   : 50%           0.000164
   : 75%           0.000507
   : max           0.081541
   : Name: percent, dtype: float64

1.  Make a *horizontal* boxplot of the first 10 `percent` values of the
    data set (remember that you need to import the graphics library and
    its plotting module):



In [1]:
<<baby>>
plt.boxplot(baby[:10]['percent'],vert=False)
plt.show()

1.  Make a *scatterplot* of the `percent` values: for the x-argument of the
    `plt.scatter` function use the `index`, which is stored in `baby.index`:



In [1]:
<<baby>>
plt.scatter(baby.index,baby.percent)
plt.show()

1.  It's very hard to see any trends with so many data points! Plot
    again but this time limit the number of data points to 100:



In [1]:
<<baby>>
plt.scatter(baby.index[:100],baby.percent[:100])
plt.show()

1.  You can clearly see how a few names are very popular while a bunch
    of other names are not. To really test trends, you'd have to
    randomize your sample though (a look at the original dataset shows
    that you only picked the boy names in the year 1880.

2.  Print the functions that you used with the corresponding magic
    command.
    
        %whos

3.  Lastly, save a Python (`.py`) version of your code to your PC.

